packages/flutter/lib/src/widgets/tap_and_drag_gestures.dart

```diff
@@ -1026,8 +1008,8 @@ class TapAndDragGestureRecognizer extends OneSequenceGestureRecognizer with _Tap
       // has been accepted and it has moved past the [slopTolerance] but has not moved
       // a sufficient global distance from the initial position to be considered a drag.
       // In this case since the gesture cannot be a tap, it defaults to a drag.
-
-      _pastSlopTolerance = _pastSlopTolerance || slopTolerance != null && _getGlobalDistance(event, _initialPosition) > slopTolerance!;
+      final double computedSlop = computeHitSlop(event.kind, gestureSettings);
+      _pastSlopTolerance = _pastSlopTolerance || _getGlobalDistance(event, _initialPosition) > computedSlop;
```

Comment:
```dart
      } else if (_dragState == _DragState.possible) {
        if (_start == null) {
          // Only check for a drag if the start of a drag was not already identified.
          _checkDrag(event);
        }
        // This can occur when the recognizer is accepted before a [PointerMoveEvent] has been
        // received that moves the pointer a sufficient global distance to be considered a drag.
        if (_start != null) {
          _acceptDrag(_start!);
        }
      }
```
this should change to:
```dart
      } else if (_dragState == _DragState.possible) {
        if (_start == null) {
          // Only check for a drag if the start of a drag was not already identified.
          _checkDrag(event);
        }
        assert(_start == null)
      }
```
If the `_checkDrag` successed, `_acceptDrag` will called immediately

Reply:
Ohh, I understand what the case is, you are right ：）